twisted/conch/ssh/keys.py

```diff
@@ -229,7 +302,17 @@ def _fromString_PUBLIC_OPENSSH(cls, data):
         @rtype: L{twisted.conch.ssh.keys.Key}
         @raises BadKeyError: if the blob type is unknown.
         """
-        blob = base64.decodestring(data.split()[1])
+        #EC keys can come in multiple formats.
+        if data.startswith('ecdsa'):
+            newKey = cls(load_ssh_public_key(data, default_backend()))
+            newKey.ecKeyName = data[:19]
+            return newKey
+        elif 'ecdsa' in data:
+            # This string doesn't seem to be supported by any of cryptgraphy.io's utility functions.
+            # So do it the long way by grabbing the bytes we need and passing them on.
+            blob = re.search(".*?nist.\d{3}(.*)", data, re.DOTALL).group(1)
```

Comment:
It strikes me that doing a regex over the entire key blob might be the source of some of your unreliability.  Surely the octets `nist` might appear at some point in the key that has nothing to do with the type.